pkg/mcs/resourcemanager/server/token_buckets.go

```diff
@@ -419,21 +423,47 @@ func (gtb *GroupTokenBucket) updateTokens(now time.Time, burstLimit int64, clien
 }
 
 // request requests tokens from the corresponding slot.
-func (gtb *GroupTokenBucket) request(now time.Time,
+func (gtb *GroupTokenBucket) request(
+	now time.Time,
 	requiredToken float64,
 	targetPeriodMs, clientUniqueID uint64,
 ) (*rmpb.TokenBucket, int64) {
 	burstLimit := gtb.getBurstLimit()
 	gtb.updateTokens(now, burstLimit, clientUniqueID, requiredToken)
 	slot, ok := gtb.tokenSlots[clientUniqueID]
 	if !ok {
-		return &rmpb.TokenBucket{Settings: &rmpb.TokenLimitSettings{BurstLimit: burstLimit}}, 0
+		return &rmpb.TokenBucket{
+			Settings: &rmpb.TokenLimitSettings{BurstLimit: burstLimit},
+			Tokens:   0.0,
+		}, 0
 	}
 	res, trickleDuration := slot.assignSlotTokens(requiredToken, targetPeriodMs)
 	// Update bucket to record all tokens.
 	gtb.Tokens -= slot.lastTokenCapacity - slot.tokenCapacity
 	slot.lastTokenCapacity = slot.tokenCapacity
-
+	// Verify whether the allocated token is invalid, such as negative values, math.Inf, or math.NaN.
+	if res.Tokens <= 0 || math.IsInf(res.Tokens, 0) || math.IsNaN(res.Tokens) {
+		// Output detailed information to help locate the problem.
+		log.Warn("assigned token is invalid",
+			zap.Time("now", now),
+			zap.String("resource-group-name", gtb.resourceGroupName),
+			zap.Uint64("client-unique-id", clientUniqueID),
+			zap.Uint64("target-period-ms", targetPeriodMs),
+			zap.Float64("required-token", requiredToken),
+			zap.Float64("assigned-tokens", res.Tokens),
+			zap.Int("slot-len", len(gtb.tokenSlots)),
+			zap.Float64("fill-rate", float64(slot.fillRate)),
+			zap.Int64("burst-limit", slot.burstLimit),
+			zap.Float64("require-tokens-sum", slot.requireTokensSum),
+			zap.Float64("token-capacity", slot.tokenCapacity),
```

Comment:
I suspect it's because this field turned into 'nan', but it's hard for me to guess how it happened. Can we add a separate log specifically for when this field becomes 'nan'?